src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -345,6 +346,7 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
 
             // DCP stuff
             _innerBuilder.Services.AddSingleton<IDcpExecutor, DcpExecutor>();
+            _innerBuilder.Services.AddSingleton<IConsoleLogsService, DcpExecutor>();
```

Comment:
This doesn't do what you think. It'll create 2 instances. You want to do this https://github.com/dotnet/aspire/blob/05e04413c795d2ee10708061ef3dcf847b53930c/src/Aspire.Hosting/DistributedApplicationBuilder.cs#L329-L330

Doesn't need to be a hosted service but you get what I mean 😄